src/s1reader/s1_reader.py

```diff
@@ -627,6 +640,96 @@ def get_ascending_node_time_orbit(orbit_state_vector_list: ET,
     return anx_time_orbit
 
 
+def get_osv_list_from_orbit(orbit_file: str | list,
+                            swath_start: datetime.datetime,
+                            swath_stop: datetime.datetime):
+    '''
+    Get the list of orbit state vectors as ElementTree (ET) objects from the orbit file `orbit_file`
+
+    - When `orbit_file` is a list, then all of the OSV lists are
+      merged and sorted. The merging is done by concatenating
+      the OSVs to the "base OSV list".
+    - The "base OSV list" is the one that covers
+      the swath's start / stop time with padding applied,
+      so that the equal time spacing is guaranteed during that time period.
+
+    Parameters
+    ----------
+    orbit_file: str | list
+        Orbit file name, or list of the orbit file names
+
+    swath_start: datetime.datetime
+        Sensing start time of the swath
+    swath_stop: datetime.datetime
+        Sensing stop time of the swath
+
+    Returns
+    -------
+    orbit_state_vector_list: ET
+        Orbit state vector list
+    '''
+    if isinstance(orbit_file, str):
+        orbit_tree = ET.parse(orbit_file)
+        orbit_state_vector_list = orbit_tree.find('Data_Block/List_of_OSVs')
+        return orbit_state_vector_list
+
+    elif isinstance(orbit_file, list) and len(orbit_file):
+        # Concatenate the orbit files' OSV lists
+
+        padding = datetime.timedelta(seconds=PADDING_TIME_ORBIT)
+        n_orbit_file = len(orbit_file)
+
+        orbit_start_time_list = []
+        orbit_stop_time_list = []
+
+        for filename in orbit_file:
+            start_time_str = \
+                os.path.basename(filename).split('_')[6].replace('V','')
+            orbit_start_time_list.append(datetime.datetime.strptime(start_time_str,
+                                                                    orbitfile_datetime_format))
+            stop_time_str = \
+                os.path.basename(filename).split('_')[7].replace('V','').replace('.EOF','')
+            orbit_stop_time_list.append(datetime.datetime.strptime(stop_time_str,
+                                                                   orbitfile_datetime_format))
```

Comment:
we should have some function that gets the start and/or stop time elsewhere in the code because it's so common

Reply:
Considering this as resolved as I've removed the code above.